### PR TITLE
Check for all playback engines 

### PR DESCRIPTION
**Pull request by paulbalaji**
After checking for a Unity installation, check for all expected playback engines

Also includes drive-by fix for adding blank project to README 

Relates to: [UTY-1437](https://improbableio.atlassian.net/browse/UTY-1437)

### CODE REVIEW

File: scripts/prepare-unity.sh [hunk 1]
```diff
@@ -37,11 +37,22 @@ export GOPATH
...
-  exit 0
 else
   echo "Could not find a Unity $UNITY_PINNED_VERSION installation."
 fi
 
+if [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/LinuxStandaloneSupport" ] || \
+   [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/AndroidPlayer" ] || \
+   [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/iOSSupport" ] || \
+   [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/WindowsStandaloneSupport" ] || \
+   [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/MacStandaloneSupport" ]
```
Comment (ghost) [inline scripts/prepare-unity.sh]: For Mac this path is different for MacStandaloneSupport, also the directories should be:

`"$IMPROBABLE_UNITY_VERSIONED/Editor/Data/PlaybackEngines/..."`

Comment (paulbalaji) [reply scripts/prepare-unity.sh]: yes you're right - the paths here are the mac ones

will add the windows/mac changes 👍

File: scripts/prepare-unity.sh [hunk 2]
```diff
@@ -37,11 +37,22 @@ export GOPATH
...
+then
+  echo "All PlaybackEngines are already installed in: $IMPROBABLE_UNITY_VERSIONED/PlaybackEngines."
+  exit 0
+else
+  echo "Could not find all playback engines."
```
Comment (jamiebrynes7) [inline scripts/prepare-unity.sh]: What happens if we get here because of a missing playback engine and try to reinstall all of Unity?

Comment (paulbalaji) [reply scripts/prepare-unity.sh]: then it re-installs all of Unity with all the modules specified above - did a test run on an agent and it appears to reinstall fine

File: scripts/prepare-unity.sh [hunk 3]
```diff
@@ -41,13 +41,19 @@ else
...
+  PLAYBACK_ENGINE_ROOT="$IMPROBABLE_UNITY_VERSIONED/Editor/Data/PlaybackEngines"
+elif isMacOS; then
+  PLAYBACK_ENGINE_ROOT="$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines"
+fi
+
+if [ -d "$PLAYBACK_ENGINE_ROOT/LinuxStandaloneSupport" ] && \
+   [ -d "$PLAYBACK_ENGINE_ROOT/AndroidPlayer" ] && \
+   [ -d "$PLAYBACK_ENGINE_ROOT/iOSSupport" ] && \
+   [ -d "$PLAYBACK_ENGINE_ROOT/WindowsStandaloneSupport" ] && \
+   [ -d "$PLAYBACK_ENGINE_ROOT/MacStandaloneSupport" ]
```
Comment (ghost) [inline scripts/prepare-unity.sh]: `    [ -d "$PLAYBACK_ENGINE_ROOT/MacStandaloneSupport" ] ` could still be invalid for Mac Editor, AFAIK it should be installed within the .app - unless our installer may do a redundant build support on top?

Comment (paulbalaji) [reply scripts/prepare-unity.sh]: yup - on mac the `MacStandaloneSupport` directory is inside the .app at `/Applications/Unity/Unity/Contents/PlaybackEngines/MacStandaloneSupport`

in which case I should probably not check for this directory when setting up it's a mac agent

File: scripts/prepare-unity.sh [hunk 4]
```diff
@@ -37,11 +37,28 @@ export GOPATH
 
 if [ -d "$IMPROBABLE_UNITY_VERSIONED" ]; then
   echo "Unity is already installed in:" $IMPROBABLE_UNITY_VERSIONED
-  exit 0
 else
   echo "Could not find a Unity $UNITY_PINNED_VERSION installation."
 fi
 
+if isWindows; then
```
Comment (ghost) [inline scripts/prepare-unity.sh]: IMO these new additions can go within the `if [ -d "$IMPROBABLE_UNITY_VERSIONED" ]; then`

**Review by gkassabli — APPROVED** [review]
Lgtm in general, but restructuring code to avoid checking modules if Unity isn't installed would be nice

**Review by ghost — APPROVED** [review]

File: scripts/prepare-unity.sh [hunk 5]
```diff
@@ -37,7 +37,24 @@ export GOPATH
...
+    PLAYBACK_ENGINE_ROOT="$IMPROBABLE_UNITY_VERSIONED/Editor/Data/PlaybackEngines"
+  elif isMacOS; then
+    PLAYBACK_ENGINE_ROOT="$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines"
+  fi
+
+  if [ -d "$PLAYBACK_ENGINE_ROOT/LinuxStandaloneSupport" ] && \
+     [ -d "$PLAYBACK_ENGINE_ROOT/AndroidPlayer" ] && \
+     [ -d "$PLAYBACK_ENGINE_ROOT/iOSSupport" ] && \
+     [ -d "$PLAYBACK_ENGINE_ROOT/WindowsStandaloneSupport" ] && \
+     [ isMacOS || -d "$PLAYBACK_ENGINE_ROOT/MacStandaloneSupport" ]
```
Comment (jamiebrynes7) [inline scripts/prepare-unity.sh:52]: Why does Mac gotta be a special snowflake

**Review by jamiebrynes7 — APPROVED** [review]

File: scripts/prepare-unity.sh (hunk 5 above)
Comment (gkassabli) [reply scripts/prepare-unity.sh:52]: Mac has it's own support installed in the Unity app package itself instead of standalone playback engine

Comment (jared-improbable) [reply scripts/prepare-unity.sh:52]: A comment to that effect would be very useful.

**Review by jared-improbable — APPROVED** [review]
Single comment aside, LGTM

File: scripts/prepare-unity.sh (hunk 5 above)
Comment (paulbalaji) [reply scripts/prepare-unity.sh:52]: will add that 👍

**Pull request merged**